### ISSUE TITLE
Expose channel features to RPC

### ISSUE THREAD

**Issue by niftynei** [ElementsProject/lightning#3678]
`channel_announcement` encapsulates a 4th, possibly distinct, set of features for a node/link layer, but afaict currently there's no way to view/query them from the json-rpc.

**Comment by darosior** [comment]
I use `listchannels` on the scid.

**Comment by niftynei** [comment]
what field in `listchannels` are you using?

**Comment by darosior** [comment]
channel_flags iirc
-------- Original Message --------

**Comment by niftynei** [comment]
Ah yes. `channel_flags` is a `channel_update` field; this issue refers to `features` on `channel_announcement`.

**Comment by darosior** [comment]
Ok, checked today and it's not `channel_flags`, I think I confused and I in fact used `listnodes` (hence no `channel_announcement`).

**Comment by niftynei** [comment]
Yes, I'm aware of the other features. I ran into this issue when attempting to verify adding `channel` flags for glightning.

**Comment by sangbida** [comment]
This issue is being closed as part of issue cleanup.

**Reason**: Very old issue (5.8 years old) no recent activity

If you believe this issue should remain open, please comment and we can reopen it.
